### PR TITLE
Support multiple tax years in UI, move forms to ustaxes-forms

### PR DESCRIPTION
**Pull request by zakpatterson**
**What kind of change does this PR introduce?** 
- Feature

A redo of #566

Forms are now contained in [ustaxes-forms](//github.com/ustaxes/ustaxes-forms). Tax years will be modeled using separate branches of that repo, and each branch will be added here as a git submodule.

Added scripts to set up submodules for 2020 and 2021. 

Added the UI to enable building 2021 returns.

Data models and validations have been moved to [ustaxes-core](//github.com/ustaxes/ustaxes-core). So our structure now is:

```
             ustaxes-core
   /                 |       \
ustaxes-forms-2020   |     ustaxes-forms-2021
  \     (forms are single repo, multi branch)
   \                 |      / 
    \                |     / 
                ustaxes
```

Note: Setting up this structure did require updating NPM. This does not work on NPM 6. `npm ci` will also fail for users running npm6, and there is no way to warn them, since `ci` doesn't run any of the `preinstall`/`prepare` tasks before updating `node_modules`. We display an error in front of `start` or `test`.

About complexity, of course this adds a ton. If a user needs to add a new piece of UI for a new piece of user data, and update the data model, and update a tax form with that new piece of data, that will be three pull requests. On the other hand, UI PRs can't make updates to the data model or tax forms, and can't break tests in those projects. So it ...is also good? Perhaps time will tell.

### CODE REVIEW

**Comment by thegrims** [comment]
recording

Filing status should throw an error and not allow progression to the next page until it is filled out
<details><summary>Browser metadata</summary>





```
Path:      /spouseanddependent
Browser:   Chrome 95.0.4638.54 on Windows 10
Viewport:  1536 x 754 @1.25x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-662--ustaxes.netlify.app%2Fspouseanddependent&os=Windows&os_version=10&browser=Chrome&browser_version=95.0&resolution=1536x754&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-662--ustaxes.netlify.app/spouseanddependent) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=617bc9538a8eb50008ddb08b&commentID=617c3d39e0b1ae4e86b2d314&resolution=resolved)

**Comment by thegrims** [comment]
State wages seems to sometimes throw an error when not filled out, sometimes not

recording


<details><summary>Browser metadata</summary>





```
Path:      /income/w2jobinfo
Browser:   Chrome 95.0.4638.54 on Windows 10
Viewport:  1536 x 754 @1.25x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-662--ustaxes.netlify.app%2Fincome%2Fw2jobinfo&os=Windows&os_version=10&browser=Chrome&browser_version=95.0&resolution=1536x754&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-662--ustaxes.netlify.app/income/w2jobinfo) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=617bc9538a8eb50008ddb08b&commentID=617c3e2c10a9216198220bba&resolution=resolved)

**Comment by thegrims** [comment]
gotcha

**Comment by ctSkennerton** [comment]
It doesn't look like `CURRENT_YEAR` from `src/data/federal.ts` is changed in this PR, it's still a constant of 2020.

**Comment by zakpatterson** [comment]
> It doesn't look like `CURRENT_YEAR` from `src/data/federal.ts` is changed in this PR, it's still a constant of 2020.

Great catch @ctSkennerton, it was a bit annoying to remove the constant, but I was able to.

File: src/data/questions.ts [hunk 1]
```diff
@@ -79,7 +78,7 @@ export const questions: Question[] = [
   ),
   q(
     'LIVE_APART_FROM_SPOUSE',
-    `Did you lived apart from your spouse for all of ${CURRENT_YEAR}?`,
+    `Did you lived apart from your spouse for all of the year?`,
```
Comment (thegrims) [inline src/data/questions.ts:81]: Should be _Have you lived apart from your spouse for the current tax year?_ rather than _Did you lived_

Comment (zakpatterson) [reply src/data/questions.ts]: Going with "did you live" because it's the period it's talking about doesn't include the present.

**Comment by thegrims** [comment]
> State wages seems to sometimes throw an error when not filled out, sometimes not
> 
> recording
> 
> Browser metadata
> [Open Deploy Preview](https://deploy-preview-662--ustaxes.netlify.app/income/w2jobinfo) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=617bc9538a8eb50008ddb08b&commentID=617c3e2c10a9216198220bba&resolution=resolved)

@zakpatterson Do you have any ideas on this?

**Comment by zakpatterson** [comment]
> > State wages seems to sometimes throw an error when not filled out, sometimes not
> > recording
> > Browser metadata
> > [Open Deploy Preview](https://deploy-preview-662--ustaxes.netlify.app/income/w2jobinfo) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=617bc9538a8eb50008ddb08b&commentID=617c3e2c10a9216198220bba&resolution=resolved)
> 
> @zakpatterson Do you have any ideas on this?

I tried to have a minimal diff in pages such as this one, if you look at the diff the only changes are a couple of imports.

The field itself has `required={false}`, and the error you're seeing is when a value does not match the expected pattern for a currency. So I suspect there is a value in that field which is not being displayed because the value is non-numeric.

Is it possible your form filler is inputting non-numeric data into the value field?

**Comment by zakpatterson** [comment]
Good catch on the state wages field, added as a new issue #701

**Comment by zakpatterson** [comment]
@thegrims do you know what netlify needs? There is a permission error when cloning the submodule.

**Comment by zakpatterson** [comment]
I've finally convinced myself this strategy is a bad one. See #778

**Comment by zakpatterson** [comment]
This is replaced by #778

**Pull request closed without merging**